lms/djangoapps/grades/tasks.py

```diff
@@ -0,0 +1,54 @@
+"""
+This module contains tasks for asynchronous execution of grade updates.
+"""
+
+from celery import task
+from django.contrib.auth.models import User
+
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from lms.djangoapps.courseware.courses import get_course_by_id
+from opaque_keys.edx.keys import UsageKey
+from opaque_keys.edx.locator import CourseLocator
+from openedx.core.djangoapps.content.block_structure.api import get_course_in_cache
+
+from .config.models import PersistentGradesEnabledFlag
+from .transformer import GradesTransformer
+from .new.subsection_grade import SubsectionGradeFactory
+
+
+@task()
+def recalculate_subsection_grade(user_id, course_id, usage_id):
```

Comment:
I definitely second having a separate modular method for updating the course grade.

However, the assumed design is that whenever a problem is submitted, both the containing subsection grade and the course grade gets updated.  It wouldn't make sense to enqueue a separate async celery task for updating the course grade - when the subsection grade is already being updated in a celery task.  So this should happen within the same task.